R/read_slf.R

```diff
@@ -3,185 +3,212 @@
-
-  return(slf)
 }
 
 #' Read a Source Linkage episode file
 #'
-#' @inheritParams read_slf
-#' @inheritParams fst::read_fst
-#'
-#' @return a [tibble][tibble::tibble-package]
+#' @inherit read_slf
 #' @export
 #'
 #' @examples
 #' \dontrun{
 #' read_slf_episode("1718",
-#'   columns = c("anon_chi", "dob", "demographic_cohort"),
-#'   from = 100000,
-#'   to = 101000
+#'   col_select = c("anon_chi", "dob", "demographic_cohort")
 #' )
 #'
 #' read_slf_episode(c("1718", "1819"),
-#'   columns = c("anon_chi", "dob", "demographic_cohort"),
-#'   from = 100000,
-#'   to = 101000
+#'   col_select = c("anon_chi", "dob", "demographic_cohort"),
+#'   as_data_frame = FALSE
 #' )
 #' }
-read_slf_episode <-
-  function(year,
-           columns = NULL,
-           partnerships = NULL,
-           recids = NULL,
-           ...) {
-    # TODO add option to drop blank CHIs?
-    # TODO add a filter by recid option
-    return(
-      read_slf(
-        year = year,
-        file_version = "episode",
-        partnerships = unique(partnerships),
-        recids = unique(recids),
-        columns = unique(columns),
-        ...
-      )
+read_slf_episode <- function(
+    year,
+    col_select = NULL,
+    partnerships = NULL,
+    recids = NULL,
+    as_data_frame = TRUE,
+    dev = TRUE,
+    columns = lifecycle::deprecated()) {
+  if (lifecycle::is_present(columns)) {
+    lifecycle::deprecate_soft(
+      "0.10.0",
+      "read_slf_episode(columns)",
+      "read_slf_episode(col_select)"
     )
+    col_select <- columns
   }
-
+  # TODO add option to drop blank CHIs?
+  # TODO add a filter by recid option
+  return(
+    read_slf(
+      year = year,
+      col_select = unique(col_select),
+      file_version = "episode",
+      partnerships = unique(partnerships),
+      recids = unique(recids),
+      as_data_frame = as_data_frame,
+      dev = dev
+    )
+  )
+}
 
 #' Read a Source Linkage individual file
 #'
-#' @inheritParams read_slf
-#' @inheritParams fst::read_fst
+#' @inherit read_slf
 #'
-#' @return a [tibble][tibble::tibble-package]
 #' @export
 #'
 #' @examples
 #' \dontrun{
 #' read_slf_individual("1718",
-#'   columns = c("anon_chi", "dob", "hri_scot"),
-#'   from = 100000, to = 101000
+#'   col_select = c("anon_chi", "dob", "hri_scot")
 #' )
 #' read_slf_individual(c("1718", "1819"),
-#'   columns = c("anon_chi", "dob", "hri_scot"),
-#'   from = 100000,
-#'   to = 101000
+#'   col_select = c("anon_chi", "dob", "hri_scot"),
+#'   as_data_frame = FALSE
 #' )
 #' }
-read_slf_individual <-
-  function(year,
-           columns = NULL,
-           partnerships = NULL,
-           ...) {
-    return(
-      read_slf(
-        year = year,
-        file_version = "individual",
-        partnerships = unique(partnerships),
-        columns = unique(columns),
-        ...
-      )
+read_slf_individual <- function(
+    year,
+    col_select = NULL,
+    partnerships = NULL,
+    as_data_frame = TRUE,
+    dev = TRUE,
```

Comment:
```suggestion
    dev = FALSE,
```
Should default be false here?